teek/_widgets/treeview.py

```diff
@@ -0,0 +1,452 @@
+import functools
+from collections.abc import MutableSequence, Iterable
+
+import teek
+from teek._widgets.base import Widget, ChildMixin
+from teek._structures import ConfigDict
+from teek._tcl_calls import make_thread_safe
+
+
+class FallbackConfigDict(ConfigDict):
+    """
+    Config dictionary which uses fallback values if the given handler
+    function cannot be executed
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        # Handler functions
+        self._handlers = {}
+
+        # Fallback values (used if handler function cannot be executed)
+        self._fallbacks = {}
+
+    def _set(self, option, value):
+        try:
+            # Set value of fallback
+            self._fallbacks[option] = value
+            # Look for handler function(s)
+            handlers = (self._handlers[option] if option in self._handlers
+                        else self._handlers['*'])
+
+            if not isinstance(handlers, Iterable):
+                handlers = [handlers]
+
+            for handler in handlers:
+                handler(None, option, value)
+        except Exception:
```

Comment:
Why are you using `except Exception` here? What kinds of exceptions are you guarding against?

Reply:
> I caught the generic expression for keeping the `FallbackConfigDict` usable for other purposes where another exception (not AttributeError) is raised.

In this case, an `AttributeError` is raised but the fallback value should also be returned if another exception is thrown. It is not the best solution to catch a generic exception but for keeping it usable for other code which raises another kind of exception.

Reply:
Resolved by replacing the `FallbackConfigDict`.